### PR TITLE
Fix DropZone class names on drop

### LINKED ISSUE #65788: Dropzone re-renders for previously triggered media upload dropzone
### Description

The previously used dropzones for media upload in image blocks are being shown when dropping other images.

### Step-by-step reproduction instructions

1. Open Site Editor
2. Insert at least two images and a cover block (easy to see in a grid of images)
3. Open the media inserter sidebar
4. Drag and drop to replace one image
5. Drag to replace a second image
6. See that the first image shows the dropzone even when replacing the second image

### Screenshots, screen recording, code snippet

image

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes

### PR DESCRIPTION
**Pull request by ajlende**
## What?

Removes the `is-dragging-over-element` classname on drop.

## Why?

Fixes https://github.com/WordPress/gutenberg/issues/65788.

## How?

Removes the `is-dragging-over-element` on the `dragend` event as `dragleave` isn't called on drop.

## Testing Instructions

1. Insert at least two images and a cover block (easy to see in a grid of images)
2. Open the media inserter sidebar
3. Drag and drop to replace one image
4. Drag to replace a second image
5. See that only the expected dropzone overlay is shown.

## Screenshots or screencast 

https://github.com/user-attachments/assets/d14d2d3b-5e24-40fe-8b2a-5bbeb4ad0a94

### CODE REVIEW

File: packages/components/src/drop-zone/index.tsx [hunk 1]
```diff
@@ -116,8 +117,6 @@ export function DropZoneComponent( {
 			( ( type === 'file' && onFilesDrop ) ||
 				( type === 'html' && onHTMLDrop ) ||
 				( type === 'default' && onDrop ) ),
-		'has-dragged-out': ! isDraggingOverElement,
-		// Keeping the following classnames for legacy purposes
```
Comment (ajlende) [inline packages/components/src/drop-zone/index.tsx:120]: The `.has-dragged-out` class was added in #62044, but it seems to work fine with `.is-dragging-over-document` that existed prior to #62044. Maybe @tyxla or @ciampo know something more?

I assume the classes that existed before were utility classes for consumers to add additional styling, so I removed the comment too.

It should be safe to remove as it's a recent change for 6.7 and isn't used anywhere else.

**Review by jeryj — APPROVED** [review]
Working well for me. Fixes the duplicate drop zone overlay issue. Maybe worth waiting to merge until @tyxla or @ciampo chime in about the removal of `has-dragged-out` and the related comment.

File: packages/components/CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@
 -   `ToggleGroupControl`: Fix arrow key navigation in RTL ([#65735](https://github.com/WordPress/gutenberg/pull/65735)).
 -   `ToggleGroupControl`: indicator doesn't jump around when the layout around it changes ([#65175](https://github.com/WordPress/gutenberg/pull/65175)).
 -   `Composite`: fix legacy support for the store prop ([#65821](https://github.com/WordPress/gutenberg/pull/65821)).
+-   `DropZone`: fix class names on drop ([#65798](https://github.com/WordPress/gutenberg/pull/65798))
```
Comment (tyxla) [inline packages/components/CHANGELOG.md]:
```suggestion
-   `DropZone`: fix class names on drop ([#65798](https://github.com/WordPress/gutenberg/pull/65798)).
```

**Review by tyxla — APPROVED** [review]
Works well in my testing and makes sense to be rid of an extra classname, especially when we have so many others in place 👍 

Thanks @ajlende!

**Comment by ajlende** [comment]
Manual cherry-pick opened in https://github.com/WordPress/gutenberg/pull/65924

**Pull request merged**